modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/PartitionReplicaLifecycleManager.java

```diff
@@ -1703,4 +1703,9 @@ public HybridTimestamp currentSafeTimeForZonePartition(int zoneId, int partId) {
         return requireNonNull(zoneResourcesManager.getZonePartitionResources(new ZonePartitionId(zoneId, partId))).raftListener()
                 .currentSafeTime();
     }
+
+    @TestOnly
+    public CompletableFuture<ZonePartitionReplicaListener> zonePartitionResourcesReplicaFuture(ZonePartitionId zonePartitionId) {
+        return zoneResourcesManager.getZonePartitionResources(zonePartitionId).replicaListenerFuture();
```

Comment:
Could you please extract the whole `getZonePartitionResources` method instead? I will need it in my other PR as well.

Reply:
Ok. I will do that. My goal was to preserve package-private access to `ZonePartitionResources` class.